apps/electron/layers/main/src/db/secondary-db.ts

```diff
@@ -74,13 +67,12 @@ export class SecondaryWorkspaceSQLiteDB extends BaseSQLiteAdapter {
 
   // wrap the fn with connect and close
   // it only works for sync functions
-  run = (fn: () => void) => {
+  run = async (fn: () => Promise<void>) => {
     try {
       if (this.runCounter === 0) {
-        this.connect();
```

Comment:
`run` was designed so that db operations inside will release DB connection after use. I am not sure where is secondary DB is connected after the change?